lib/golomb/basicFilter.js

```diff
@@ -0,0 +1,14 @@
+'use strict';
+
+const Golomb = require('./golomb');
+const {U64} = require('n64');
+
+class BasicFilter extends Golomb {
+  constructor() {
+    super();
+    this.m = new U64(784931);
```

Comment:
https://github.com/bcoin-org/bcoin/blob/c1cd7ac0171e4125552b58d87e689b255bc9a9e9/lib/golomb/golomb.js#L261
`this.m` is actually not being used. golomb.js instead uses the global const `M` with the value fixed to 784931. I think it would be a good idea to pass `M` as a parameter to the gcsfilter instead of `m` and let the constructor set `this.m=this.M`.